haystack/connector/crawler.py

```diff
@@ -98,32 +100,32 @@ def crawl(self,  output_dir: Union[str, Path, None] = None,
         if not output_dir.exists():
             output_dir.mkdir(parents=True)
 
+        crawled_data: list = []
         is_not_empty = len(list(output_dir.rglob("*"))) > 0
         if is_not_empty and not overwrite_existing_files:
             logger.info(
                 f"Found data stored in `{output_dir}`. Delete this first if you really want to fetch new data."
             )
-            return []
         else:
             logger.info(f"Fetching from {urls} to `{output_dir}`")
-
-            filepaths = []
-
             sub_links: Dict[str, List] = {}
 
             # don't go beyond the initial list of urls
             if crawler_depth == 0:
-                filepaths += self._write_to_files(urls, output_dir=output_dir)
+                crawled_data += self._write_to_files(urls, output_dir=output_dir)
             # follow one level of sublinks
             elif crawler_depth == 1:
                 for url_ in urls:
                     existed_links: List = list(sum(list(sub_links.values()), []))
                     sub_links[url_] = list(self._extract_sublinks_from_url(base_url=url_, filter_urls=filter_urls,
                                                                      existed_links=existed_links))
                 for url in sub_links:
-                    filepaths += self._write_to_files(sub_links[url], output_dir=output_dir, base_url=url)
+                    crawled_data += self._write_to_files(sub_links[url], output_dir=output_dir, base_url=url)
+
+        if return_documents:
```

Comment:
How about moving the conversion of documents to the `run()` method? It'd simplify the `crawl()` method to always return file paths.

Reply:
@oryx1729 I did this change to provide the same interface for **run** and **crawl** method for end-user. If you think it better option then I will refactor the code.